examples/with-typescript2/package.json

```diff
@@ -0,0 +1,20 @@
+{
+  "name": "with-typescript2",
```

Comment:
I think we can either remove the old example and use this or name it `with-typescript-plugin`, what do you think @timneutkens?

Reply:
Maybe maintain the old one for compatibility with old Next.js versions?

Reply:
replace with-typescript2 -> with-typescript-plugin